src/Compilers/CSharp/Test/Semantic/Semantics/ObjectAndCollectionInitializerTests.cs

```diff
@@ -508,10 +677,21 @@ System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
     #endregion
 }
 ";
-            CreateStandardCompilation(source).VerifyDiagnostics(
-                // (7,41): error CS1061: 'MyList<string>' does not contain a definition for 'Add' and no extension method 'Add' accepting a first argument of type 'MyList<string>' could be found (are you missing a using directive or an assembly reference?)
-                //         var coll = new MyList<string> { "str" };
-                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, @"""str""").WithArguments("MyList<string>", "Add").WithLocation(7, 41));
+            string expectedOperationTree = @"
+IObjectCreationExpression (Constructor: MyList<System.String>..ctor()) (OperationKind.ObjectCreationExpression, Type: MyList<System.String>, IsInvalid) (Syntax: 'new MyList< ... > { ""str"" }')
+  Initializer: IObjectOrCollectionInitializerExpression (OperationKind.ObjectOrCollectionInitializerExpression, Type: MyList<System.String>, IsInvalid) (Syntax: '{ ""str"" }')
+      Initializers(1): IInvocationExpression ( ? MyList<System.String>.Add()) (OperationKind.InvocationExpression, Type: ?, IsInvalid) (Syntax: '""str""')
+          Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: MyList<System.String>) (Syntax: 'MyList<string>')
+          Arguments(1): IArgument (ArgumentKind.Explicit, Matching Parameter: null) (OperationKind.Argument, IsInvalid) (Syntax: '""str""')
+              ILiteralExpression (OperationKind.LiteralExpression, Type: System.String, Constant: ""str"") (Syntax: '""str""')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS1061: 'MyList<string>' does not contain a definition for 'Add' and no extension method 'Add' accepting a first argument of type 'MyList<string>' could be found (are you missing a using directive or an assembly reference?)
+                //         var coll = /*<bind>*/new MyList<string> { "str" }/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, @"""str""").WithArguments("MyList<string>", "Add").WithLocation(7, 51)
```

Comment:
Operation unit test generator doesn't seem to include the diagnostic location in the generated comment for expected diagnostics. Filed https://github.com/dotnet/roslyn/issues/20788 to track updating the generator.